### PR TITLE
Setup Boilerplate Code

### LINKED ISSUE #1: Housekeeping - Repository Setup
This issue will be used to track all setup tasks that need to take place before work on user stories can begin.

- [x] Create Develop Branch
- [x] Issue Templates (User Stories)
- [x] Backend Boilerplate
  - [x] ASP.NET Core API Project
  - [x] xUnit Test Project
- [x] Frontend Boilerplate
  - [x] Next.js
  - [x] ESLint + Airbnb Preset
  - [x] Material UI
- [x] CI/CD

### PR DESCRIPTION
**Pull request by miosenpai**
This PR completes all the code related tasks in #1 which includes setting up backend boilerplate, frontend boilerplate, and CI/CD.

### CODE REVIEW

**Review by TianYueDong — APPROVED** [review]
looks good

File: Buddies.API/WeatherForecast.cs
Comment (andy9775) [inline Buddies.API/WeatherForecast.cs]: Perhaps remove some of these boilerplate files. Or at least rename them to something relevant and keep it empty.

File: Buddies.Tests/UnitTest1.cs
Comment (andy9775) [inline Buddies.Tests/UnitTest1.cs]: Again, remove some of the default boilerplate.

File: Buddies.Web/.eslintrc.json
Comment (andy9775) [inline Buddies.Web/.eslintrc.json:12]: Generally, most people leave the react import in - it's a good habit to get into. But I'll leave it up to you.

File: Buddies.Web/src/styles/Home.module.css
Comment (andy9775) [inline Buddies.Web/src/styles/Home.module.css]: I don't think this css (file) is applied anywhere is it?

File: Buddies.sln
Comment (andy9775) [inline Buddies.sln:31]: I'm not too familiar with Visual Studio, is it best practice to check this file in?

File: .github/workflows/testing.yml
Comment (andy9775) [inline .github/workflows/testing.yml:7]: What's the workflow here? Will each member clone their own repo and issue a PR from that? As an FYI, if you're working on open source projects this tends to be the workflow - only one person (Tech lead) tends to have write access and everyone else issues pr's, even if they work in the same company.

File: Buddies.API/Controllers/WeatherForecastController.cs
Comment (andy9775) [inline Buddies.API/Controllers/WeatherForecastController.cs]: Same comment on the boiler plate code.

**Review by andy9775 — CHANGES_REQUESTED** [review]
I would rename the directories. Instead of `Buddies.API` just `server` or `api` and instead of `Buddies.Web` just `client`.

**Comment by miosenpai** [comment]
> I would rename the directories. Instead of `Buddies.API` just `server` or `api` and instead of `Buddies.Web` just `client`.

According to [.NET guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-namespaces), using `Product.Feature` is a recommended scheme for separating parts within a product. This scheme is also used in industry projects such as [GitHub's Action Runner](https://github.com/actions/runner/tree/main/src).

File: Buddies.sln
Comment (miosenpai) [reply Buddies.sln:31]: Yes it is best practice to check this file in, the `.gitignore` file at the root of the repository was generated through the .NET SDK and already excludes all unnecessary files from Visual Studio/.NET development.

File: Buddies.Web/.eslintrc.json
Comment (miosenpai) [reply Buddies.Web/.eslintrc.json:12]: Next.js auto imports React, so this rule will not be necessary ([Next.js's ESLint plugin](https://github.com/vercel/next.js/blob/3dc4b524e79754804b8e53bd1783967d48ce5d13/packages/eslint-config-next/index.js#L18) also disables this, although we had to extend from a different config due to [documented incompatibilities](https://nextjs.org/docs/basic-features/eslint#recommended-plugin-ruleset)).

File: .github/workflows/testing.yml
Comment (miosenpai) [reply .github/workflows/testing.yml:7]: We will create separate branches from `develop` for each user story and implementation of user stories will take place in their respective branches. When a story is close to completion, one of the members working on said story will PR its branch to `develop`, and the PR will be merged once it passes review. At the end of a monthly phase, `develop` will be merged into `main` (the branch where CD to AWS will be set up), signifying a "release".

We plan to briefly document this workflow in the repo's README.md that will be written in the next few days (#5).

**Review by andy9775 — APPROVED** [review]
LGTM!

**Pull request merged**
